tests/datasource/fluent/conftest.py

```diff
@@ -249,26 +274,56 @@ def _post_fake_db_datasources_callback(
     url = request.url
     logger.info(f"{request.method} {url}")
 
-    item = _CLOUD_API_FAKE_DB.get(url, MISSING)
-    if request.body and item is MISSING:
-        payload = json.loads(request.body)
```

Comment:
This `cloud_api_mock` logic was bad, it wasn't doing any kind of check to see if a dasource of the same name had already been `POST`ed.